indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerConfig.java

```diff
@@ -316,6 +316,14 @@ public HadoopyShardSpec getShardSpec(Bucket bucket)
     return schema.getTuningConfig().getShardSpecs().get(bucket.time).get(bucket.partitionNum);
   }
 
+  /**
+   * Job instance should have Configuration set (by calling {@link #addJobProperties(Job)}
```

Comment:
This was my main concern regarding forcing addJobProperties.  It does not seem incorrect to addInputPaths without adding job properties (e.g. "Use defaults" case).

If we were sure that any jobProperties configured in the spec should have highest precedence, addInputPaths could do this directly...  Then the "Use Defaults" case is handled by not overriding anything in the Spec.